core/src/main/resources/error/error-classes.json

```diff
@@ -3,6 +3,18 @@
     "message" : [ "Field name %s is ambiguous and has %s matching fields in the struct." ],
     "sqlState" : "42000"
   },
+  "COLUMN_ALIASES_NOT_ALLOWED_IN_OPERATION" : {
+    "message" : [ "Columns aliases are not allowed in %s." ],
+    "sqlState" : "10000"
```

Comment:
See https://github.com/apache/spark/pull/33560 for the standard SQLSTATEs. I think this would be 42000; we'll do another pass before the 3.3 release to clean up any incorrect ones.